### ISSUE TITLE
Unable to build with Qt6.8

### ISSUE THREAD

**Issue by txtsd** [kotatogram/kotatogram-desktop#464]
### Steps to reproduce

1. Try to build with Qt6.8

### Expected behaviour

It should build

### Actual behaviour

It throws errors about the `#error` macro being removed in Qt6.8

These errors show up even when all the `#error` macro lines are replaced with `static_assert()`.

### Operating system

Arch Linux

### Version of Kotatogram Desktop

1.5.0.4.r1005.g0581eb6

### Installation source

Other (unofficial) source: git

**Comment by ilya-fedin** [comment]
A tip: you can still build with Qt 5

**Comment by txtsd** [comment]
> A tip: you can still build with Qt 5

Yes, definitely, but I'm in charge of packaging https://aur.archlinux.org/packages/kotatogram-desktop-git so this is here as an upstream tracker.

**Comment by ilya-fedin** [comment]
> These errors show up even when all the `#error` macro lines are replaced with `static_assert()`.

This #error is in Qt sources, you have to replace them in Qt, not kotato

> Yes, definitely, but I'm in charge of packaging https://aur.archlinux.org/packages/kotatogram-desktop-git so this is here as an upstream tracker.

I'm highly doubt it will be solved any time soon, so switch the package to Qt 5 for the time being? So that people can still use it at least.

**Comment by txtsd** [comment]
> > These errors show up even when all the `#error` macro lines are replaced with `static_assert()`.
> 
> This #error is in Qt sources, you have to replace them in Qt, not kotato

Ah I should've realised!

> > Yes, definitely, but I'm in charge of packaging https://aur.archlinux.org/packages/kotatogram-desktop-git so this is here as an upstream tracker.
> 
> I'm highly doubt it will be solved any time soon, so switch the package to Qt 5 for the time being? So that people can still use it at least.

That's not a bad idea. I will do it soon.

**Comment by dreirund** [comment]
> > This `#error` is in Qt sources, you have to replace them in Qt, not kotato
> 
> *Ah I should've realised!*

By the way: Is there any upstream report to Qt already? If so, where?

> > I'm highly doubt it will be solved any time soon, so switch the package to Qt 5 for the time being? So that people can still use it at least.
> 
> *That's not a bad idea. I will do it soon.*

Looking forward to it.

Regards!

**Comment by ilya-fedin** [comment]
> By the way: Is there any upstream report to Qt already? If so, where?

No. Do you think they will revert the change? It looks pretty intentional.

**Comment by dreirund** [comment]
> > By the way: Is there any upstream report to Qt already? If so, where?
> 
> *No. Do you think they will revert the change? It looks pretty intentional.*

I understood [the comment posted by @ilya-fedin](#issuecomment-2528102579):

> This #error is in Qt sources, you have to replace them in Qt, not kotato

that it is an Qt issue (Qt sources beeing inconsistend with themselved).

So, it is not a Qt issue but a kotatogtam issue?


Regards!

**Comment by ilya-fedin** [comment]
Qt basically removed the API kotatogram is using and returns a error on an attempt to use it when building. From my understanding, that's not an issue at all, just incompatibility, there seem to be no guarantee by Qt that sources using it will build without changes with next versions. There seem to be only binary guarantees (you can build the sources with the Qt version it was made for and then update Qt, it should still work).

In other words, you can build only with old Qt and that seem to be absolutely intended.
